presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/optimization/JdbcComputePushdown.java

```diff
@@ -100,18 +110,23 @@ public PlanNode visitFilter(FilterNode node, Void context)
             TableHandle oldTableHandle = oldTableScanNode.getTable();
             JdbcTableHandle oldConnectorTable = (JdbcTableHandle) oldTableHandle.getConnectorHandle();
 
+            RowExpression toTranslate = expressionOptimizer.optimize(node.getPredicate(), OPTIMIZED, session);
+            toTranslate = logicalRowExpressions.convertToConjunctiveNormalForm(toTranslate);
+            TranslatedExpression<JdbcSql> jdbcSqlTranslatedExpression = translateWith(
+                    toTranslate,
+                    rowExpressionToSqlTranslator,
+                    oldTableScanNode.getAssignments());
+
             // TODO: remove dependency on oldTableLayoutHandle, currently it needs oldTableLayoutHandle to get predicate
-            if (!oldTableHandle.getLayout().isPresent()) {
+            if (!oldTableHandle.getLayout().isPresent() || !jdbcSqlTranslatedExpression.getTranslated().isPresent()) {
```

Comment:
You could do finer granularity checking if `jdbcSqlTranslatedExpression.getTranslated()` is not present. A possible way is to traverse the translated expression by walking through `AND`/`OR` to decide which part can be pushed down.

Reply:
Adding TODO.